src/infrastructure/figma/figma-service.ts

```diff
@@ -403,20 +416,24 @@ export class FigmaService {
 	): Promise<AtlassianDesign | null> =>
 		this.withErrorTranslation(async () => {
 			try {
-				const fileResponse = await figmaClient.getFile(
-					fileKey,
-					{
-						ids: [nodeId],
-						depth: 0, // Exclude children of the target node to avoid a massive response payload and high network latency.
-						node_last_modified: true,
-					},
-					credentials.accessToken,
-				);
+				const [fileResponse, fileMetaResponse] = await Promise.all([
+					figmaClient.getFile(
+						fileKey,
+						{
+							ids: [nodeId],
+							depth: 0, // Exclude children of the target node to avoid a massive response payload and high network latency.
+							node_last_modified: true,
+						},
+						credentials.accessToken,
+					),
+					figmaClient.getFileMeta(fileKey, credentials.accessToken),
+				]);
 
 				return tryTransformNodeToAtlassianDesign({
 					fileKey,
 					nodeId,
 					fileResponse,
+					fileMetaResponse,
 				});
 			} catch (e) {
 				if (e instanceof NotFoundHttpClientError) return null;
```

Comment:
non-blocking: is this still catching properly / returning null after the update to use `Promsise.all()`? idr if there's a specific unit/integration test to check this path

(ditto for any other `Promise.all()` changes in this PR)

Reply:
There are tests for checking if it returns null for this code path.
This block still catches exceptions thrown  even after adding `Promise.all` and the type is not affected.